### DISCUSSION TITLE
TypeError: Cannot assign to read-only property 'current'

### DISCUSSION THREAD

**Discussion by pb-sarulatha-annadurai** [software-mansion/react-native-gesture-handler#3071] [Q&A]
I am trying  to setState in onFinalise() to update the current position, though I wrap it with runOnJs the app crashed with the following error. 

![simulator_screenshot_ABC05F25-28DF-4221-8E16-3511B4A6A690](https://github.com/user-attachments/assets/f8673df1-3679-4e17-ab7a-c1a5e16abdd2)

Below is the code 

```
const panGesture = Gesture.Pan()
    .enabled(!props.disabled)
    .onFinalize(() => {
      const currentThumb = activeThumb.value

      // function to  setstate of current position
      runOnJS(currentThumb.onPanStop)()
    })
```

**Comment by latekvo** [comment]
Hi @imsarulatha

It's not entirely clear what `onPanStop` does, and so it's difficult to tell what could be causing this error.
Could you please provide us with a minimum reproduction example?
